### DISCUSSION TITLE
A mysterious error I cant explain: TypeError: Element type "Request" must be followed by either attribute specifications, ">" or "/>".

### DISCUSSION THREAD

**Discussion by GunnarPiho** [nextgenhealthcare/connect#5826] [Q&A]
I have a very simple write-to-file channel. 

As soon as I add even an empty transformer to the destination, I will get the following transformer error.

Transformer error
ERROR MESSAGE: Error evaluating transformer
com.mirth.connect.server.MirthJavascriptTransformerException: 
CHANNEL:	Simple channel
CONNECTOR:	To file
SCRIPT SOURCE:	
SOURCE CODE:	
94:         }
95:         eval('importPackage(' + Packages.java.lang.Class.forName(className).getPackage().getName() + ')');
96:     }
97: }
98: function doScript() {
99: msg = new XML(connectorMessage.getTransformedData());
100: if (msg.namespace('') != undefined) { default xml namespace = msg.namespace(''); } else { default xml namespace = ''; }
101: function doFilter() { phase[0] = 'filter'; return true; }function doTransform() { phase[0] = 'transformer'; logger = Packages.org.apache.log4j.Logger.getLogger(phase[0]);
102: 
103: 
LINE NUMBER:	99
DETAILS:	TypeError: Element type "Request" must be followed by either attribute specifications, ">" or "/>".
	at 55d46997-2f86-4181-bb37-5e608b15c934_JavaScript_Filter_Transformer_1:99 (doScript)
	at 55d46997-2f86-4181-bb37-5e608b15c934_JavaScript_Filter_Transformer_1:128
	at com.mirth.connect.server.transformers.JavaScriptFilterTransformer$FilterTransformerTask.doCall(JavaScriptFilterTransformer.java:235)
	at com.mirth.connect.server.transformers.JavaScriptFilterTransformer$FilterTransformerTask.doCall(JavaScriptFilterTransformer.java:187)
	at com.mirth.connect.server.util.javascript.JavaScriptTask.call(JavaScriptTask.java:114)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)

**Comment by jonbartels** [comment]
This is the internal code from Mirth trying to parse something as XML and that XML has an error.

What data types are you actually needing to process with your channel?

What are the data types for the channel set to?

**Reply by pacmano1** [reply]
And what is the outbound message template set to?

**Reply by GunnarPiho** [reply]
Yes, XML tag, but as much I know, I do not use XML in a particular channel. I have a really simple channel. I just installed Mirth 4.3.0 in my Windows 11 for testing. Both inbound and outbound data types are HL7 v2.x. The source is the channel reader, and the destination is the file writer. If I have neither filters nor transformers, all works fine with the following message.

MSH|^~\&|ADM|Sending Organization|ALL|Receiving Organization|||QBP^E22^QBP_E22||D|2.7
QPD|E22^Authorization 
Request^CIHI0003||^^^ISO^PHN|Everywoman^Mary^Patrick^^^^L^|19680120|MSP^|EXT
RCP|I

If I add an empty transformer, I will get the error when using the same message. The same also happens when I add the following step

tmp = msg;
tmp['MSH']['MSH.7']['MSH.7.1'] = Now("yyyyMMddhhmmss");
tmp['MSH']['MSH.10']['MSH.10.1'] = UUIDGenerator.getUUID().substr(14);

**Reply by pacmano1** [reply]
Can you post an export of the channel that causes this?

**Reply by jonbartels** [reply]
`DETAILS: TypeError: Element type "Request" must be followed by either attribute specifications, ">" or "/>".`

and

`QPD|E22^Authorization
Request^CIHI0003||^^^ISO^PHN|Everywoman^Mary^Patrick^^^^L^|19680120|MSP^|EXT`

I bet there is a stray character or missing escape sequence. 

Export the  HL7 message as a file also.

**Reply by GunnarPiho** [reply]
[Query Sender JMS.zip](https://github.com/nextgenhealthcare/connect/files/11885243/Query.Sender.JMS.zip)
The channel export file is attached

**Reply by GunnarPiho** [reply]
@jonbartels, thanks, this was the reason. Solved.
